### PR TITLE
ci: reduce usage

### LINKED ISSUE #1798: Reduce CI/CD usage
We should reduce the CI/CD usage by running some workflow only for some changed files.

This is not related to any pricing limitation, but simply make the workflows more reactive as we do have limitations on the number of concurrent workflows/jobs, and reduce some CPU usage (its free but still eats energy).

I propose to split the test workflow into multiple workflow per component (analyzer/api/api-client/playout/legacy/...) so we can trigger some test only if some files changed.

Having workflow per component would move the CI/CD in a monorepo like structure, which seems to fit our needs.

We can use reusable workflow to reduce duplication. I already have a working example here https://github.com/jooola/libretime/pull/86

This does mean a total rebuild of our test workflows.

### PR DESCRIPTION
**Pull request by jooola**
Fixes #1798

Please squash!

### CODE REVIEW

**Comment by jooola** [comment]
The required workflow names changed.

File: .github/workflows/legacy.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: Legacy
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - .github/workflows/**
+      - api/**
```
Comment (paddatrapper) [inline .github/workflows/legacy.yml:8]: Why run on api changes? Currently there are no tests that would verify things breaking in legacy if the API changed

Comment (jooola) [reply .github/workflows/legacy.yml:8]: I expect both to be tightly coupled together during the transition to django. We already add more coupling with #1644 or #1787 this was more a way to make sure we don't break legacy when changing something on the api side. 

I can remove this path though.

Comment (paddatrapper) [reply .github/workflows/legacy.yml:8]: Not necessary with #1644. With that, we will need to test both

**Comment by paddatrapper** [comment]
I've actually dropped the required tests - in general all of our tests are required for merging. We need to check if that does what is expected for auto-merging though

**Pull request merged**